### PR TITLE
Fix #14184 - HTTPS/HTTP cookie separation issue (aka Failed to set session cookie. Maybe you are using HTTP instead of HTTPS)

### LINKED ISSUE #14184: Failed to set session cookie. Maybe you are using HTTP instead of HTTPS
### Steps to reproduce
1. Install PMA 4.8.0
2. Login via HTTP

### Expected behaviour
After entering the login information, I should be able to log in

### Actual behaviour
It doen't log me in and fails with this error : **Failed to set session cookie. Maybe you are using HTTP instead of HTTPS** 
In previous version of PMA 4.7.9 I didn't face this issue I was able to log in via HTTP and HTTPS without any issue.
Now I am only able to login via HTTPS.

How can I disable this as I cannot find anything in the docs, the closest thing I was able to find is this [Link](https://docs.phpmyadmin.net/en/latest/setup.html?#http-authentication-mode)

I am really confused about this.

I checked the file `common.inc.php` where this error is mentioned, but still wasn't able to figure out the issue.

### Web Server configuration

      server{

		listen	   2002;
		listen	   2004;
		server_name  localhost;
		
		# The Document Root
		root	/custom/path;
		
		rewrite "/sess([0-9a-zA-Z]{16})/(.*)" /$2;
		
		# Redirect server error pages to the static pages
		error_page  404					/nginx/404.html;
		error_page  497					/nginx/497.html;
		error_page  500 502 503 504		/nginx/50x.html;		
		location ~ (index.php|/)$ {
		
			fastcgi_pass	unix:/usr/local/php/var/fpm9178.sock;
			if ($uri ~ "^/phpmyadmin/") {
				fastcgi_pass	unix:/usr/local/php/var/fpm9179.sock;
			}
			fastcgi_read_timeout 3600;
			fastcgi_index	index.php;
			fastcgi_param	SCRIPT_FILENAME  $document_root$fastcgi_script_name;
			fastcgi_param	SCRIPT_NAME	  $fastcgi_script_name;
			include			fastcgi_params;
		}
	
		location ~ \.php$ {
			fastcgi_pass	unix:/usr/local/php/var/fpm9178.sock;
			if ($uri ~ "^/phpmyadmin/") {
				fastcgi_pass	unix:/usr/local/php/var/fpm9179.sock;
			}

			fastcgi_read_timeout 3600;
			fastcgi_index	index.php;
			fastcgi_param	SCRIPT_FILENAME	$document_root$fastcgi_script_name;
			fastcgi_param	SCRIPT_NAME		$fastcgi_script_name;
			include			fastcgi_params;
		}
		
		client_max_body_size 50M;
	}

### Server configuration
**Operating system**:
CentOS and Ubuntu 

**Web server:**
Nginx 1.12.2

**Database:**
MySQL 5.5.59

**PHP version:**
PHP 5.6.35

**phpMyAdmin version:**
4.8.0

### Client configuration
**Browser:**
Chrome, FireFox

**Operating system:**
Windows

### PR DESCRIPTION
**Pull request by williamdes**
The issue is: if cookies are set with Secure flag they will be available to be sent only on HTTPS.
If you login by https and then try to login by http this will fail. (after this PR, not anymore)
But if you login on http and then go to https everything works well since the cookie was not defined with Secure flag.

Fix #14184 - HTTPS/HTTP cookie separation issue (aka Failed to set session cookie. Maybe you are using HTTP instead of HTTPS)

Duplicate-of-pull-request: #14420 (closed by myself some time ago)
Fixes: #14184

### Description
Fixes #14184

### Screenshots
image

### CODE REVIEW

**Comment by williamdes** [comment]
I assigned you all because I think this is an important change, it changes the cookie name from `phpMyAdmin` to `phpMyAdmin_http` for HTTP (this change could be removed) and `phpMyAdmin` to `phpMyAdmin_http` for HTTPS

**Comment by williamdes** [comment]
This PR #15273 will not solve the issue : the session expired and so the token is invalid when submitting the form, reloading and submitting the form another time fixes the issue

**Review by MauricioFauth — APPROVED** [review]

**Comment by williamdes** [comment]
@ibennetch If you can review this PR please I would be soo glad, It will fix part of the issue but I do not want to fix everything in this PR
Since this one is adding a new cookie name for HTTPS I need your approval to merge in QA

File: libraries/classes/Config.php [hunk 1]
```diff
@@ -1611,6 +1615,37 @@ public function setCookie($cookie, $value, $default = null,
...
+    }
+
+    /**
+     * Get the real cookie name
+     *
+     * @param string $cookieName The name of the cookie
+     * @return string
+     */
+    public function getCookieName($cookieName) {
+        return $cookieName. ( ($this->isHttps()) ? '_https' : '' );
```
Comment (williamdes) [inline libraries/classes/Config.php]: Here

**Review by MauricioFauth — APPROVED** [review]

**Pull request merged**